bundled-guava/src/main/java/org/apache/iceberg/GuavaClasses.java

```diff
@@ -94,6 +96,8 @@ public class GuavaClasses {
     Iterables.class.getName();
     CountingOutputStream.class.getName();
     Suppliers.class.getName();
+    Cache.class.getName();
+    CacheBuilder.class.getName();
```

Comment:
Usually we use the [`Caffeine` caches](https://github.com/ben-manes/caffeine) instead (which mostly have the same API as Guava's).

Is there a reason that `Caffeine` won't work here?

Reply:
Accepted, please check again